Adding Initial project with License boilerplate

keshiha96 opened the pull request:
Creating the initial base project of wso2am-k8s-operator which simply deploys the WSO2 API Manager and exposes the service. This mainly contains the artifacts folder where the configuration YAML files are defined, hack folder with files that generate the clientset, informers and listers folders, main.go file that has the controller logic and finally the controller.go that performs the actual logic and functionalities. Also, this project folder is copyrighted to WSO2 Inc. 2019, and the license boilerplate is defined for relevant files.

keshiha96 committed:
Adding Initial project with license

keshiha96 committed:
year updated in update-codegen.sh file

keshiha96 committed:
year updated in verify-codegen.sh

dinusha92 commented on artifacts/examples/crd.yaml:
```diff
@@ -0,0 +1,11 @@
+  names:
+    kind: Apimanager
+    plural: apimanagers
+  scope: Namespaced
```
At a new line here.

keshiha96 replied:
done

dinusha92 commented on controller.go:
```diff
@@ -0,0 +1,524 @@
+	}
+}
+
+func newService(apimanager *apimv1alpha1.Apimanager) *corev1.Service {
```
Shall we add a method comment here.

keshiha96 replied:
sure

keshiha96 committed:
inserted space

keshiha96 committed:
added new line

keshiha96 committed:
added new line

keshiha96 committed:
added new line

keshiha96 committed:
added new line

keshiha96 committed:
added comment for service

inserted comment for describing newService, replaced foo object with apimanager in few comments
